contracts/utils/ExternalMulticall.sol

```diff
@@ -24,13 +24,26 @@ abstract contract ExternalMulticall is IExternalMulticall {
     /// of the batched calls reverts
     /// @param targets Array of target addresses of batched calls
     /// @param data Array of calldata of batched calls
+    /// @param values Array of values to be sent with each call
     /// @return returndata Array of returndata of batched calls
     function externalMulticall(
         address[] calldata targets,
-        bytes[] calldata data
-    ) public virtual override returns (bytes[] memory returndata) {
+        bytes[] calldata data,
+        uint256[] calldata values
+    ) public payable virtual override returns (bytes[] memory returndata) {
         uint256 callCount = targets.length;
-        require(callCount == data.length, "Parameter length mismatch");
+        require(
+            callCount == data.length && callCount == values.length,
+            "Parameter length mismatch"
+        );
+        uint256 value = 0;
+        for (uint256 i = 0; i < callCount; ) {
+            value += values[i];
+            unchecked {
+                ++i;
+            }
+        }
+        require(msg.value >= value, "Insufficient value");
```

Comment:
We should require an exact match here because the opposite is guaranteed to be a user error